### PR TITLE
Return an error object inline with the JSON RPC spec.

### PR DESCRIPTION
**Pull request by tcoulter**
We were previously just returning an error message, which breaks the rpc spec and web3.

### CODE REVIEW

**Comment by kumavis** [comment]
I kept your changes in case we prefer this behaviour but opted to just pass the error on instead - if this is a problem lets discuss

**Comment by tcoulter** [comment]
I don't think that's up to spec. According to the JSON RPC spec, you return an object with an error property. Web3 is angry if anything else is returned.

**Comment by kumavis** [comment]
that makes sense.
though angry is good, its an error after all.
in development ive wanted it to fail as early as possible, makes debugging easier
hmmm i wonder if we are caching error responses...

**Comment by tcoulter** [comment]
Let me rephrase. Web3 says "Invalid JSON RPC response" instead of showing the actual error. So without this PR, we're effectively squashing errors.

**Comment by kumavis** [comment]
ok thats a big deal
havent been testing against web3

**Comment by kumavis** [comment]
published as 4.0.1
